packages/frint-cli/commands/new.js

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable no-multi-spaces, no-use-before-define */
+
+const mkdirp = require('mkdirp');
+const request = require('request');
+const tar = require('tar');
+
+const createApp = require('frint').createApp;
+
+const DEFAULT_ORGANIZATION = 'Travix-International';
+const DEFAULT_REPOSITORY = 'frint';
+const DEFAULT_BRANCH = 'master';
+const DEFAULT_EXAMPLES_DIR = 'examples';
+const DEFAULT_EXAMPLE = 'counter';
+const DESCRIPTION_TEXT = `
+Usage:
+
+  $ frint new
+  $ frint new <directory>
+  $ frint new <directory> --example <example>
+
+Example:
+
+  $ frint new myapp --example kitchensink
+  $ frint new myapp --example frint-vue/tree/master/examples/basic
+
+You can find a list of all available official examples here:
+https://github.com/Travix-International/frint/tree/master/examples
+`.trim();
+const INVALID_EXAMPLE_ARG_TEXT = `
+Invalid <example> value. Must be in one of the following formats:
+
+  * <name>
+  * <organization>/<repository>/tree/<branch>/**
+`.trim();
+
+module.exports = createApp({
+  name: 'new',
+  providers: [
+    {
+      name: 'summary',
+      useValue: 'Scaffolds a new Frint app in specified directory',
+    },
+    {
+      name: 'description',
+      useValue: DESCRIPTION_TEXT,
+    },
+    {
+      name: 'execute',
+      useFactory: function useFactory(deps) {
+        return function execute() {
+          deps.console.log('Initializing...');
+          Promise.resolve(deps)
+            .then(mapDepsToContext)
+            .then(createOutputDirectory)
+            .then(streamExampleToOutputDirectory)
+            .then(() => deps.console.log('Done!'))
+            .catch(deps.console.error);
+        };
+      },
+      deps: [
+        'console',
+        'params',
+        'pwd',
+      ],
+    }
+  ],
+});
+
+function mapDepsToContext(deps) {
+  return new Promise((resolve, reject) => {
+    const ctx = {};
+
+    // The <example> param has two shapes:
+    // * <name> - example name from the official Frint GitHub repository
+    // * <organization>/<repo>/tree/<branch>/** - full GitHub path to an arbitrary example
+    const example = deps.params.example || DEFAULT_EXAMPLE;
+    if (!example.match(/(^(\w|-)+$)|^\/?(\w|-)+(\/(\w|-)+){3,}\/?$/)) {
+      reject(INVALID_EXAMPLE_ARG_TEXT);
+    }
+
+    // Split by '/' and filter out empty results.
+    // <example> arg might start or end with a separator.
+    const exampleParts = example.split('/').filter(str => str !== '');
+
+    const isFullExamplePath = exampleParts.length > 1;
+
+    ctx.organization = isFullExamplePath ? exampleParts[0]                 : DEFAULT_ORGANIZATION;
```

Comment:
please extract the `ctx` generation into a separate function.

```js
function getContextOrWhatever(example) {
  if (example.indexOf('/') === -1) {
    return exampleFromDefaultRepo;
  }

  return exampleFromCustomRepo;
}
```

Doing so would also help us avoid multiple ternaries below.